File: src/main/java/com/aliiensmp/aliienResize/Economy/currencyOptions/Experience.java
package com.aliiensmp.aliienResize.Economy.currencyOptions;

import com.aliiensmp.aliienResize.Economy.CurrencyProvider;
import org.bukkit.entity.Player;

public class Experience implements CurrencyProvider {

    private final String suffix;

    /**
     * Creates the experience currency adapter.
     *
     * @param suffix suffix displayed beside prices
     */
    public Experience(String suffix) {
        this.suffix = suffix;
    }

    /**
     * Rounds a configured price up to the minimum whole number of levels required.
     *
     * @param amount configured price
     * @return required player levels
     */
    private int getRequiredLevels(double amount) {
        return Math.max(0, (int) Math.ceil(amount));
    }

    @Override
    public boolean isValid() {
        return true;
    }

    @Override
    public boolean hasBalance(Player player, double amount) {
        return player.getLevel() >= getRequiredLevels(amount);
    }

    @Override
    public boolean withdraw(Player player, double amount) {
        if (!hasBalance(player, amount)) return false;

        player.setLevel(player.getLevel() - getRequiredLevels(amount));
        return true;
    }

    @Override
    public String suffix() {
        return suffix;
    }
}

File: src/main/kotlin/com/aliiensmp/aliienResize/Economy/currencyOptions/Experience.kt
package com.aliiensmp.aliienResize.Economy.currencyOptions

import com.aliiensmp.aliienResize.Economy.CurrencyProvider
import org.bukkit.entity.Player
import kotlin.math.ceil

class Experience(override val suffix: String) : CurrencyProvider {

    override val isValid: Boolean get() = true

    private fun getRequiredLevels(amount: Double): Int {
        return maxOf(0, ceil(amount).toInt())
    }

    override fun hasBalance(player: Player, amount: Double): Boolean {
        return player.level >= getRequiredLevels(amount)
    }

    override fun withdraw(player: Player, amount: Double): Boolean {
        if (!hasBalance(player, amount)) return false

        player.level -= getRequiredLevels(amount)
        return true
    }
}